mmv1/third_party/terraform/services/artifactregistry/data_source_artifact_registry_docker_image.go

```diff
@@ -0,0 +1,331 @@
+package artifactregistry
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+// https://cloud.google.com/artifact-registry/docs/reference/rest/v1/projects.locations.repositories.dockerImages#DockerImage
+type DockerImage struct {
+	name           string
+	uri            string
+	tags           []string
+	imageSizeBytes string
+	mediaType      string
+	uploadTime     string
+	buildTime      string
+	updateTime     string
+}
+
+func DataSourceArtifactRegistryDockerImage() *schema.Resource {
+
+	return &schema.Resource{
+		Read: DataSourceArtifactRegistryDockerImageRead,
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Project ID of the project.`,
+			},
+			"repository": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The repository name.`,
+			},
+			"region": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The location of the artifact registry repository. For example, "us-west1".`,
+			},
+			"image": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The image name to fetch.`,
+			},
+			"digest": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `The image digest to fetch.  This cannot be used if tag is provided.`,
```

Comment:
Could you use `ConflictsWith: []string{"tag"},` instead? Same in reverse for `tag`.

Reply:
That's a good catch!  I'm not sure how I forgot that.

Reply:
@roaks3 Ok, pushed a commit to address this.